Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/NativeLibrary.java

```diff
@@ -401,18 +406,64 @@ private NativeLibrary()
 		CacheClassesAndMethods();
 	}
 
-	public static void displayAlertMsg(final String alert)
+	public static boolean displayAlertMsg(final String caption, final String text, final boolean yesNo)
 	{
-		Log.error("[NativeLibrary] Alert: " + alert);
+		Log.error("[NativeLibrary] Alert: " + text);
 		final EmulationActivity emulationActivity = sEmulationActivity.get();
-		if (emulationActivity != null)
+		boolean result = false;
+		if (emulationActivity == null)
 		{
-			emulationActivity.runOnUiThread(() -> Toast.makeText(emulationActivity, "Panic Alert: " + alert, Toast.LENGTH_LONG).show());
+			Log.warning("[NativeLibrary] EmulationActivity is null, can't do panic alert.");
+		}
+		else if (yesNo)
+		{
+			result = displayYesNo(emulationActivity, caption, text);
 		}
 		else
 		{
-			Log.warning("[NativeLibrary] EmulationActivity is null, can't do panic toast.");
+			emulationActivity.runOnUiThread(() -> Toast.makeText(emulationActivity, "Panic Alert: " + text, Toast.LENGTH_LONG).show());
 		}
+		return result;
+	}
+
+	private static boolean alertResult = false;
+	private static boolean displayYesNo(final Context context, final String caption, final String text)
+	{
+		alertResult = false;
+		// Create a Handler, which allows us to effectively block whilst waiting for the user's input, then throws on completion, allowing us to exit the loop.
```

Comment:
This commit is pretty long, so please break it into two lines instead of one.

Reply:
Done.